lib/read_only/lfric/read_only.f90

```diff
@@ -0,0 +1,835 @@
+    subroutine PreEnd(this)
+        implicit none
+        class(read_only_verify_PSyDataType), intent(inout), target :: this
+    end subroutine PreEnd
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is called after the instrumented region has been
+    !! executed. After this call the value of variables after the instrumented
+    !! region will be provided. This subroutine sets the 'verify_checksum'
+    !! flag to true, causing all further checksum calls to verify that the
+    !! checksum has not changed. It also resets the next variable index to 1
+    !! again.
+    !! @param[inout] this The instance of the read_only_verify_PSyDataType.
+    subroutine PostStart(this)
+        implicit none
+        class(read_only_verify_PSyDataType), intent(inout), target :: this
+        this%verify_checksums = .true.
+        this%next_var_index = 1
+    end subroutine PostStart
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is called after the instrumented region has been
+    !! executed and all values of variables after the instrumented
+    !! region have been provided. No special functionality required here.
+    !! @param[inout] this The instance of the read_only_verify_PSyDataType.
+    subroutine PostEnd(this)
+        implicit none
+        class(read_only_verify_PSyDataType), intent(inout), target :: this
+        if(this%verbosity>0) &
+            print *,"PSYDATA: checked ", trim(this%module_name), &
+                    " ", trim(this%region_name)
+    end subroutine PostEnd
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine declares a scalar integer value.
+    !! @param[inout] this The instance of the read_only_verify_PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine DeclareScalarInt(this, name, value)
+        implicit none
+        class(read_only_verify_PSyDataType), intent(inout), target :: this
+        character(*), intent(in) :: name
+        integer, intent(in) :: value
+        this%count_checksums = this%count_checksums + 1
+    end subroutine DeclareScalarInt
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine either computes a checksum or compares a checksum
+    !! (depending on this%verify_checksums) for a single integer.
+    !! @param[inout] this The instance of the read_only_verify_PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine ChecksumScalarInt(this, name, value)
+        implicit none
+        class(read_only_verify_PSyDataType), intent(inout), target :: this
+        character(*), intent(in) :: name
+        integer, intent(in) :: value
```

Comment:
I don't agree with this: LFRic has 5 different integer types: 
```
  integer, parameter :: i_def        = int32       !< Default integer kind for application.
  integer, parameter :: i_native     = kind(i_val) !< Native kind for integer.
  integer, parameter :: i_long       = int64       !< Explicit long integer.
  integer, parameter :: i_halo_index = int64       !< Integer kind for the index used in halo swapping
  integer, parameter :: i_um         = int32       !< Default integer kind used by the UM.
```
Since they can all be different, we would have to supply a separate version for each ``i_`` value. On the other hand, if two ``i_`` are the same, compilation will fail because the two functions can not be differentiated.

So I think the right solution is to just provide the standard integers, independent of how LFRic assign them. If they use int*8, we will add them as int*8 (well, whatever the proper portable declaration in modern Fortran is). Note that this will be trivial if we use scripts to create the functions ... see #832 which I just added.